### PR TITLE
add Agent to documentation site

### PR DESCRIPTION
**Pull request by bensonlee5**
@mattcoganymede @joshasharpe - I'm going to spend some time on my flight integrating the agent documentation into the documentation website - so putting this placeholder here

### CODE REVIEW

**Comment by joshasharpe** [comment]
@bensonlee5 since you are working on this iteratively, can you ping when you are ready for a review?

**Comment by bensonlee5** [comment]
yup will do

**Comment by bensonlee5** [comment]
OK @joshasharpe @mattcoganymede @sbelkadi - would love to get your thoughts on this update.  

Stylistically, I think connectivity should be emphasized - and so created a "Ways to Capture Data" page to summarize how we can capture data from instruments.

On Agent specifically -

I called the "Processor" a "Normalizer" - with the idea being that the main use case is going to be to conform data coming off each instrument so that it can be processed consistently by Flows.  I was thinking that this would limit the complexity of the code that resides on different instruments - to make the code here a bit easier to manage.

I also renamed "Trigger" to "Listener".

Def more edits should come - I wonder whether we should call the File Watcher a File Watcher - since it could be confusing to someone familiar with the Browser-based file watcher.

Anyway - all up for thoughts - def very much in draft mode at this point.

**Comment by joshasharpe** [comment]
@bensonlee5 "with the idea being that the main use case is going to be to conform data coming off each instrument so that it can be processed consistently by Flows" This is only true for the file watcher agents. For cron or cloud watcher, wouldn't its be more than just a data normalizer? Because it has to interact with the equipment more?

File: docs/connectivity/Agent.mdx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+The Ganymede Agent is an executable that can be installed on lab instruments running Windows or Linux to capture instrument data systematically.  The agent can
+
+- filter and designate how to process data
+- interact with connected instruments
+- upload files into Ganymede Cloud
+- initiate data processing pipelines
+
+By installing Agents, your lab machines become an integrated part of the Ganymede system, enhancing the data management and processing capabilities in your lab.
+
+## What does an Agent consist of?
```
Comment (joshasharpe) [inline docs/connectivity/Agent.mdx:21]: This section definitely excludes capabilities of agent to listen on a cron interval or for cloud file changes.

File: docs/connectivity/Agent.mdx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+title: Instrument Agents
+displayed_sidebar: webUiSidebar
+---
+
+import { PlusCircleOutlined, 
+         SelectOutlined,
+         DiffOutlined
+       } from '@ant-design/icons';
+
+The Ganymede Agent is an executable that can be installed on lab instruments running Windows or Linux to capture instrument data systematically.  The agent can
```
Comment (sbelkadi) [inline docs/connectivity/Agent.mdx:12]: Some lab instruments do run Windows/Linux, but we should say here that the Agent is running on the attached PC.

File: docs/connectivity/Agent.mdx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+- What files are relevant for capture?
+- How should files (from potentially different instruments) be standardized for consumption (by a common flow)?
+- Where should the standardized file be directed?
+
+These questions are answered by the 3 components of an Agent, which function in sequence:
+
+- A **Listener**, which looks for relevant, new files appearing on an instrument computer
+- A **Normalizer**, which is a user-defined Python snippet that normalizes captured files into a desired, standardized format
+- An **Action**, which is the action to be performed with the normalized data.  The Agent can be configured to upload a data file, start a flow, or complete execution without communicating with Ganymede Cloud.
```
Comment (sbelkadi) [inline docs/connectivity/Agent.mdx:33]: Should rephrase the last part of this, to separate out when Agent takes a Ganymede action vs when it stops without action.

**Comment by bensonlee5** [comment]
@joshasharpe @sbelkadi - I attempted to address comments above, and also asked for @mattcoganymede to flip through this documentation.

The todos on the documentation side are on Linux installs and uninstalls; @joshasharpe - would you be able to put together some documentation on this?

File: docs/connectivity/Agent.mdx [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      src="https://ganymede-bio.mo.cloudinary.net/agent/Agent_Windows_Setup20230605.png"
+      width="500" />
+</div>
+&nbsp;
+
+- **Connection name** is the name seen by users in the instrument computer and in the Ganymede UI
+- **Variable definitions** are strings in the format "var_name=var_value", which allows users to set context variables to be used in the Normalizer
+- **Label** is a string that can be used to identify and group connections within Ganymede.  The label is visible in the Connections UI
+
+TODO: clarify how to specify variable definitions and label with examples
```
Comment (bensonlee5) [inline docs/connectivity/Agent.mdx:162]: @eriis1 @joshasharpe is there documentation on setting variables/labels for the windows agent somewhere?

File: docs/connectivity/Agent.mdx [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+The Agent can be uninstalled and associated service removed through the “Add or Remove Programs” panel from the Control Panel.
+
+<div class="text--center">
+  <img alt="Agent uninstall"
+      src="https://ganymede-bio.mo.cloudinary.net/agent/Agent_Uninstall20230615.png" 
+      width="500"
+      />
+</div>
+
+After uninstalling the Agent, the Ganymede folder will remain in the Program Files directory.  This folder can be deleted if desired.
```
Comment (bensonlee5) [inline docs/connectivity/Agent.mdx:284]: not sure if this is intended behavior or not

File: docs/connectivity/Overview.mdx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+Users can specify input parameters upon flow execution, giving greater flexibility to flows for which execution needs to vary based on user-specified input.
+
+Drag-and-drop becomes less appropriate if instrument runs generate many files, such that the operational overhead of specifying files to be executed becomes cumbersome.
+
+## Ganymede Agent
+
+An installable program is configured in the Ganymede app and downloaded to relevant instrument PCs.  Once executed, the program runs in the background, robust to computer restart.
+
+### When to use Ganymede Agent
```
Comment (joshasharpe) [inline docs/connectivity/Overview.mdx:32]: This should be updated based on @sbelkadi 's doc.

**Comment by bensonlee5** [comment]
> @bensonlee5 "with the idea being that the main use case is going to be to conform data coming off each instrument so that it can be processed consistently by Flows" This is only true for the file watcher agents. For cron or cloud watcher, wouldn't its be more than just a data normalizer? Because it has to interact with the equipment more?

I've been thinking a fair bit about this - my thought is that we should steer away from this in the near term even though the Agent is capable of handling this.  I think people will naturally push for this when we deal with machines that can take scripted commands, but most machines do not - and the ones that do would almost always have some sort of in-person component that is also required, making it tricky to debug issues.  I think we can lean into these things after the file upload / file upload + flow kickoff are reasonably stable.

File: docs/connectivity/Agent.mdx [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+
+Each Agent has 3 questions to answer:
+
+- What files are relevant for capture?
+- How should files (from different instrument instances) be standardized for consumption (by a common flow)?
+- Where should the standardized file be directed?
+
+These questions are answered by the 3 components of an Agent, which function in sequence:
+
+- A **Listener**, which looks for relevant, new files appearing on an instrument computer
```
[the remark this answers is not in the document]
Comment (bensonlee5) [reply docs/connectivity/Agent.mdx]: removed these capitalized terms

**Comment by bensonlee5** [comment]
TODO: maybe we should make a matrix with each of the file capture methods
TODO: Linux install/uninstall instructions

**Pull request merged**